mobile/components/views/Attractions/Search/Search.tsx

```diff
@@ -1,122 +1,110 @@
+const AttractionSearchScreen: React.FC<{ navigation: NavigationProps }> = ({ navigation }) => {
   const [loading, setLoading] = useState(true);
-  const [userAttractionsData, setUserAttractionsData] = useState<Attraction[]>(attractionsExamples);
+  const [userAttractions, setUserAttractions] = useState<Attraction[]>(attractionsExamples);
   const [attractions, setAttractions] = useState<Attraction[]>([]);
-  const [attractionDetails, setAttractionDetails] = useState<Attraction | {}>({});
   const [filterModalVisible, setFilterModalVisible] = useState(false);
-  const [attractionModalVisible, setAttractionModalVisible] = useState(false);
   const [filters, setFilters] = useState<Filters>({
-    category: "",
-    city: "",
-    country: "",
-    keyword: "",
-    radius: "10",
-    region: ""
+    category: '',
+    city: '',
+    country: '',
+    keyword: '',
+    radius: '10',
+    region: '',
   });
 
-  const [radiuses, setRadiuses] = useState<string[]>([]);
-  const navigation = useNavigation<NavigationProps>();
-
+  const [radii, setRadii] = useState<string[]>([]);
 
   useEffect(() => {
     const radiusData = ['5', '10', '15'];
-    setRadiuses(radiusData);
-    fetchUsersAttractions(1)
-      .then((attractions) => setUserAttractionsData(attractions))
-      .catch((error) => {
-        if (userAttractionsData === attractionsExamples) {
+    setRadii(radiusData);
+    fetchUserAttractions(1)
+      .then((fetchAttractions) => setUserAttractions(fetchAttractions))
+      .catch(() => {
+        if (userAttractions === attractionsExamples) {
           Alert.alert('Error', 'Failed getting user attractions. Loading demo data');
         } else {
           Alert.alert('Error', 'Failed getting user attractions.');
         }
       })
       .then(() => setLoading(false));
-  }, []);
-
+  }, [userAttractions]);
 
   useEffect(() => {
-    const filteredAttractions = userAttractionsData.filter(attraction => {
+    const filteredAttractions = userAttractions.filter((attraction) => {
       return (
-        (filters.category === "" || attraction.category === filters.category) &&
-        (filters.city === "" || attraction.city === filters.city) &&
-        (filters.country === "" || attraction.country === filters.country) &&
-        (filters.keyword === "" || attraction.name.toLowerCase().includes(filters.keyword.toLowerCase())) &&
-        (filters.region === "" || attraction.region.toLowerCase().includes(filters.region.toLowerCase()))
+        (filters.category === '' || attraction.category === filters.category) &&
+        (filters.city === '' || attraction.city === filters.city) &&
+        (filters.country === '' || attraction.country === filters.country) &&
+        (filters.keyword === '' ||
+          attraction.name.toLowerCase().includes(filters.keyword.toLowerCase())) &&
+        (filters.region === '' ||
+          attraction.region.toLowerCase().includes(filters.region.toLowerCase()))
```

Comment:
It look nasty, but it is not said that this is wrong

Reply:
I improved the appearance a bit